### PR TITLE
chore: send bundled telemetry at the end of onboarding session

### LINKED ISSUE #4590: Improve telemetry for onboarding process (send a bundled log)
### Is your enhancement related to a problem? Please describe

Reference: https://github.com/containers/podman-desktop/pull/4572#pullrequestreview-1706816125

We should send a bundled telemetry event rather than separate ones (one for each registerCommand), so we can see a summary of the entire process rather than each individual event.

### Describe the solution you'd like

Send a single onboarding event on failure / success, as per Tim's example:

```
.logUsage('compose.onboarding.checkComposeInstalled', {
  version:
  version-default: (true/false if they modified the default value)
  download:
  download-error: (error object, or -success true/false if we can't get that)
  system-wide: (only added if they got to this part)
  system-wide-error: (only added if they got to this part. error object, or true/false if we can't get that)
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

This PR makes changes to the onboarding process, by collecting telemetry data from the commands executed during the onboarding (through the response of commands), and sending the bundled data at the end of the onboarding session.

### What issues does this PR fix or reference?

Fixes #4590

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
AFAIK compose should not know anything about telemetry

File: extensions/compose/src/extension.ts [hunk 1]
```diff
@@ -95,6 +95,12 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
         downloaded: isDownloaded === '' ? false : true,
         version: composeVersionMetadata?.tag,
       });
+      return {
```
Comment (benoitf) [inline extensions/compose/src/extension.ts:98]: compose should not return special objects I would say

**Comment by feloy** [comment]
> AFAIK compose should not know anything about telemetry

The current code for the compose extension is using the `createTelemetryLogger` function available from the API to send telemetry directly, how is it different? 

Which part of the app is expected to deal with telemetry, `renderer`?

**Comment by benoitf** [comment]
I mean, tracking what has been done by the user is independent of the wish of the extension wanting to send telemetry events


There should not be any telemetry contract on callbacks/signature except calling explicitly telemetry methods

**Comment by lstocchi** [comment]
I wonder if we should allow to create a sort of transaction for telemetry.

So, from an extension, you can call a method begin transaction, which returns an id. You add content to that, and when you want to push it you call merge transaction. So we could reuse it in other places if needed.

For the onboarding we can say if the workflow is cancelled/skipped/completed, if there is any open transaction it gets closed and merged automatically, otherwise a generic telemetry with the steps name or anything else is sent

**Comment by feloy** [comment]
closing in favour of #4736

**Pull request closed without merging**